src/inc/testf.h

```diff
@@ -89,31 +89,24 @@ struct bao_test {
             testframework_fails);                                           \
     } while (0)
 
-#define BAO_TEST(suite, test)                                         \
-    static void func_bao_test_##suite##_##test(unsigned char*);       \
-    static void func_bao_test_TBD_##suite##_##test(void)              \
-    {                                                                 \
-        extern unsigned int testframework_tests;                      \
-        extern unsigned int testframework_fails;                      \
-        unsigned char failures = 0;                                   \
-        if (TESTF_LOG_LEVEL > 1) {                                    \
-            BAO_INFO_TAG();                                           \
-            printf("Running " #suite "\t" #test "\n");                \
-        }                                                             \
-        testframework_tests++;                                        \
-        func_bao_test_##suite##_##test(&failures);                    \
-        if (failures)                                                 \
-            testframework_fails++;                                    \
-    }                                                                 \
-    struct bao_test struct_bao_test_##suite##_##test __attribute__((  \
-        section(".testframework." #suite))) = { .suite_name = #suite, \
-        .test_name = #test,                                           \
-        .func_ptr = func_bao_test_TBD_##suite##_##test };             \
-    static void func_bao_test_##suite##_##test(unsigned char* failures)
-
-static inline void run_all();
-static inline int run_specific_test(char* suite, char* test);
-static inline int run_suite(char* suite);
-void bao_test_entry(void);
-
-#endif // BAO_TEST_H
+#define BAO_TEST(suite, test)                          \
+    void bao_test_##suite##_##test(unsigned char*);    \
+    void entry_test_##suite##_##test(void)             \
+    {                                                  \
+        extern unsigned int testframework_tests;       \
+        extern unsigned int testframework_fails;       \
+        unsigned char failures = 0;                    \
+        if (TESTF_LOG_LEVEL > 1) {                     \
+            BAO_INFO_TAG();                            \
+            printf("Running " #suite "\t" #test "\n"); \
+        }                                              \
+        testframework_tests++;                         \
+        bao_test_##suite##_##test(&failures);          \
+        if (failures)                                  \
+            testframework_fails++;                     \
+    }                                                  \
+    void bao_test_##suite##_##test(unsigned char* failures)

```

Comment:
As we are trying to change/delete the TAG **bao**, shouldn't this be changed to testf_test_##suite##_##test(unsigned char* failures) ?

Reply:
[See comment](https://github.com/bao-project/bao-tests/pull/6#discussion_r1199635057)